packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Primitives/CSharpType.cs

```diff
@@ -131,7 +131,7 @@ private static void ValidateArguments(Type type, IReadOnlyList<CSharpType> argum
             }
         }
 
-        internal CSharpType(
+        public CSharpType(
```

Comment:
Why do we want this to be public?  Isn't this why we have the factory method CreateCSharpType on TypeFactory?

If we make things like this public it get shard to pre / post visit the creation and modify it in plugins.

Reply:
@m-nash mentioned we already have https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Primitives/CSharpType.cs#L641 which should accomplish what we need. We don't need to make this public and we can close this PR assuming that method works as expected.

Reply:
yes, this works, thank you!